fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt

```diff
@@ -55,6 +56,57 @@
     </field>
   </xsl:template>
 
+  <!-- utk_mods_etd mode -->
+  <!-- utk_mods_etd mode --> 
+
+
+  <!-- JIRA TRAC-875 Define utk_mods_etd_author in Solr -->
+  <!-- the following template creates an _s field for single etd author -->
+  <!-- need to  use single_valued_hashset -->
+  <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Author') or 
+    (mods:role/mods:roleTerm='author')]" mode="utk_mods_etd">
+    <xsl:variable name="given-n" select="mods:namePart[@type='given']"/>
+    <xsl:variable name="family-n" select="mods:namePart[@type='family']"/>
+    <xsl:variable name="t-o-address" select="mods:namePart[@type='termsOfAddress']"/>
+
+     <field name="utk_etd_mods_name_author_s">
+       <xsl:choose>
+          <xsl:when test="$t-o-address!=''">
+               <xsl:value-of select="concat($given-n, ' ', $family-n, ', ', $t-o-address)"/>
+        </xsl:when>
+        <xsl:otherwise>
+              <xsl:value-of select="concat($given-n, ' ', $family-n)"/>
+        </xsl:otherwise>
+        </xsl:choose>
+    </field>
+  </xsl:template>
+
+
+  <!-- utk_mods_etd  -->
+  <!-- utk_mods_etd  -->
+
+  
+  <!-- JIRA TRAC-875 Define utk_mods_etd_author in Solr -->
+  <!-- the following template creates an _s field for single etd author -->
+  <xsl:template match="mods:mods/mods:name[(mods:role/mods:roleTerm='Author') or (mods:role/mods:roleTerm='author')]" mode="utk_ir_MODS">
```

Comment:
Do we have data where the roleTerm is lower case?

Reply:
Let's not worry about this one.

Reply:
Not addressing this one.